### PR TITLE
feat(text-input): text-input @carbon/react v11 sync

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[text-input]: sync with @carbon/react version #10233

### Description

@carbon/react syncing with `text-input` component. This component replaces the `input` references with `text-input`

@carbon/react version: https://react.carbondesignsystem.com/?path=/story/components-textinput--playground

### CODE REVIEW

File: packages/carbon-web-components/docs/form-story.mdx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Meta } from '@storybook/addon-docs/blocks';
 
 # Having components participate in form
 
-Though form elements in `carbon-web-components` (e.g. `<cds-input>`) are not native form elements like `<input>`, form elements in `carbon-web-components` have some extra APIs that align well to web/framework standards that allow those form elements to participate in form.
+Though form elements in `carbon-web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, form elements in `carbon-web-components` have some extra APIs that align well to web/framework standards that allow those form elements to participate in form.
```
Comment (kennylam) [inline packages/carbon-web-components/docs/form-story.mdx:7]:
```suggestion
Though form elements in `@carbon/web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, they have some extra APIs that align well with web/framework standards.
```

File: packages/carbon-web-components/docs/form.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Having components participate in form
 
-Though form elements in `carbon-web-components` (e.g. `<cds-input>`) are not native form elements like `<input>`, form elements in `carbon-web-components` have some extra APIs that align well to web/framework standards that allow those form elements to participate in form.
+Though form elements in `carbon-web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, form elements in `carbon-web-components` have some extra APIs that align well to web/framework standards that allow those form elements to participate in form.
```
Comment (kennylam) [inline packages/carbon-web-components/docs/form.md:3]:
```suggestion
Though form elements in `@carbon/web-components` (e.g. `<cds-text-input>`) are not native form elements like `<input>`, they have some extra APIs that align well with web/framework standards.
```

File: packages/carbon-web-components/docs/form-story.mdx [hunk 3]
```diff
@@ -24,7 +24,7 @@ button.addEventListener('click', () => {
   });
   event.formData = formData;
   form.dispatchEvent(event);
-  // Now `formData` is populated with the data in components such as `<cds-input>` or `<cds-dropdown>`, etc. in the `<form>`.
+  // Now `formData` is populated with the data in components such as `<cds-text-input>` or `<cds-dropdown>`, etc. in the `<form>`.
```
Comment (kennylam) [inline packages/carbon-web-components/docs/form-story.mdx:27]:
```suggestion
  // `formData` is populated with the data in components such as `<cds-text-input>` or `<cds-dropdown>`, etc. in the `<form>`.
```

File: packages/carbon-web-components/docs/form.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ button.addEventListener('click', () => {
   });
   event.formData = formData;
   form.dispatchEvent(event);
-  // Now `formData` is populated with the data in `<cds-input>`, etc. in the `<form>`.
+  // Now `formData` is populated with the data in `<cds-text-input>`, etc. in the `<form>`.
```
Comment (kennylam) [inline packages/carbon-web-components/docs/form.md:23]:
```suggestion
  // `formData` is populated with the data in `<cds-text-input>`, etc. in the `<form>`.
```

File: packages/carbon-web-components/src/components/text-input/text-input-skeleton.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import styles from './input.scss';
+
+/**
+ * Skeleton of number input.
+ */
+@customElement(`${prefix}-text-input-skeleton`)
+class CDSTextInputSkeleton extends LitElement {
+  /**
+   * `true` if the label should be hidden. Corresponds to the attribute with the same name.
+   */
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/text-input/text-input-skeleton.ts:22]:
```suggestion
  /**
   * Specify whether the label should be hidden, or not
   */
```

We should match these with the same desc from `@carbon/react`

File: packages/carbon-web-components/src/components/text-input/text-input-story.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+ */
+
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { boolean, select, number } from '@storybook/addon-knobs';
+import textNullable from '../../../.storybook/knob-text-nullable';
+import storyDocs from './input-story.mdx';
+import { prefix } from '../../globals/settings';
+import './text-input';
+import './text-input-skeleton';
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/text-input/text-input-story.ts:17]:
```suggestion
import './index';
```

**Review by kennylam — COMMENTED** [review]
@annawen1 Looks good! Just a few small things.

**Comment by kennylam** [comment]
Looks like deploy previews are failing because there's still a reference to the original stylesheet filename
```
Carbon Web Components/packages/carbon-web-components/src/components/text-input/input.scss doesn't exist
```

File: packages/carbon-web-components/src/components/text-input/text-input-story.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { boolean, select, number } from '@storybook/addon-knobs';
+import textNullable from '../../../.storybook/knob-text-nullable';
+import storyDocs from './input-story.mdx';
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/text-input/text-input-story.ts:14]:
```suggestion
import storyDocs from './text-input-story.mdx';
```

Comment (annawen1) [reply packages/carbon-web-components/src/components/text-input/text-input-story.ts:14]: Good catch!

File: packages/carbon-web-components/src/components/text-input/text-input-story.ts [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        'Warning message that is really long can wrap to more lines but should not be excessively long.'
+      ),
+      value: textNullable('Value of input (value)', ''),
+      onInput: action(`${prefix}-select-selected`),
+    }),
+  },
+};
+
+export default {
+  title: 'Components/TextInput',
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/text-input/text-input-story.ts:173]:
```suggestion
  title: 'Components/Text Input',
```
capital or lower case? might have some inconsistencies

**Pull request merged**